lib/agent0/examples/interactive_econ.py

```diff
@@ -0,0 +1,453 @@
+chain = LocalChain(LocalChain.Config())
+
+# %%
+# Parameters for pool initialization. If empty, defaults to default values, allows for custom values if needed
+config = InteractiveHyperdrive.Config(
+    position_duration=exp.term_seconds,
+    checkpoint_duration=60 * 60 * 24,  # 1 day
+    initial_liquidity=FixedPoint(20),
+    initial_fixed_rate=exp.starting_fixed_rate,
+    initial_variable_rate=exp.starting_variable_rate,
+    curve_fee=exp.curve_fee,
+    flat_fee=exp.flat_fee,
+    governance_fee=exp.governance_fee,
+)
+MINIMUM_TRANSACTION_AMOUNT = config.minimum_transaction_amount
+for k, v in config.__dict__.items():
+    print(f"{k:26} : {v}")
+print(f"{'term length':27}: {exp.term_days}")
+interactive_hyperdrive = InteractiveHyperdrive(chain, config)
+print(f"spot price = {interactive_hyperdrive.hyperdrive_interface.calc_spot_price()}")
+
+# %%
+# set up agents
+larry = interactive_hyperdrive.init_agent(base=FixedPoint(exp.amount_of_liquidity), name="larry")
+larry.add_liquidity(base=FixedPoint(exp.amount_of_liquidity))  # 10 million
+rob = interactive_hyperdrive.init_agent(base=FixedPoint(exp.amount_of_liquidity), name="rob")
+print(f"spot price = {interactive_hyperdrive.hyperdrive_interface.calc_spot_price()}")
+
+# %%
+# do some trades
+Max = NamedTuple("Max", [("base", FixedPoint), ("bonds", FixedPoint)])
+GetMax = NamedTuple("GetMax", [("long", Max), ("short", Max)])
+
+start_time = time.time()
+
+
+def get_max(
+    _interactive_hyperdrive: InteractiveHyperdrive, _share_price: FixedPoint, _current_base: FixedPoint
+) -> GetMax:
+    """Get max trade sizes.
+
+    Returns
+    -------
+    GetMax
+        A NamedTuple containing the max long in base, max long in bonds, max short in bonds, and max short in base.
+    """
+    _max_long_base = _interactive_hyperdrive.hyperdrive_interface.calc_max_long(budget=_current_base)
+    _max_long_shares = _interactive_hyperdrive.hyperdrive_interface.calc_shares_out_given_bonds_in_down(_max_long_base)
+    _max_long_bonds = _max_long_shares * _share_price
+    _max_short_bonds = FixedPoint(0)
+    try:  # sourcery skip: do-not-use-bare-except
+        _max_short_bonds = _interactive_hyperdrive.hyperdrive_interface.calc_max_short(budget=_current_base)
+    except:  # pylint: disable=bare-except
+        pass
+    max_short_shares = _interactive_hyperdrive.hyperdrive_interface.calc_shares_out_given_bonds_in_down(
+        _max_short_bonds
+    )
+    _max_short_base = max_short_shares * _share_price
+    return GetMax(
+        Max(_max_long_base, _max_long_bonds),
+        Max(_max_short_base, _max_short_bonds),
+    )
+
+
+# sourcery skip: avoid-builtin-shadow, do-not-use-bare-except, invert-any-all,
+# remove-unnecessary-else, swap-if-else-branches
+for day in range(exp.term_days):
+    amount_to_trade_base = FixedPoint(exp.amount_of_liquidity * exp.daily_volume_percentage_of_liquidity)
+    while amount_to_trade_base > MINIMUM_TRANSACTION_AMOUNT:
+        randnum = np.random.randint(0, 2)
```

Comment:
I think it would be easier to follow if you were more explicit about what this variable is for. For example:
```
from agent0.hyperdrive.state.hyperdrive_actions import HyperdriveActionType
available_actions = np.array([HyperdriveActionType.OPEN_LONG, HyperdriveActionType.OPEN_SHORT])
trade_type = rng.choice(available_actions, size=1)[0]
if trade_type == HyperdriveActionType.OPEN_LONG:
    ... # do long stuff
else:
    ... # do short stuff
```

Reply:
oh, wait, `#go long` does not mean `open long`... it means... `close short...? So if you don't follow my advice on the trade selection (I thought that was what you were doing) please add comments and improve variable names.

Reply:
I want to avoid referencing specific actions, since this is a directional toggle where going long includes closing short positions and going short includes closing long positions. I'm just gonna use:

```py
if rng.random() < 0.5:  # go long 50% of the time
...
else:  # go short 50% of the time
```